oauth_provider_jwt/models/oauth_provider_client.py

```diff
@@ -0,0 +1,187 @@
+        comodel_name='oauth.provider.scope', string='Data Scope',
+        domain=[('model_id.model', '=', 'res.users')],
+        help='Scope executed to add some user\'s data in the token.')
+    jwt_algorithm = fields.Selection(selection=[
+        ('HS256', 'HMAC using SHA-256 hash algorithm'),
+        ('HS384', 'HMAC using SHA-384 hash algorithm'),
+        ('HS512', 'HMAC using SHA-512 hash algorithm'),
+        ('ES256', 'ECDSA signature algorithm using SHA-256 hash algorithm'),
+        ('ES384', 'ECDSA signature algorithm using SHA-384 hash algorithm'),
+        ('ES512', 'ECDSA signature algorithm using SHA-512 hash algorithm'),
+        ('RS256', 'RSASSA-PKCS1-v1_5 signature algorithm using SHA-256 hash '
+         'algorithm'),
+        ('RS384', 'RSASSA-PKCS1-v1_5 signature algorithm using SHA-384 hash '
+         'algorithm'),
+        ('RS512', 'RSASSA-PKCS1-v1_5 signature algorithm using SHA-512 hash '
+         'algorithm'),
+        ('PS256', 'RSASSA-PSS signature using SHA-256 and MGF1 padding with '
+         'SHA-256'),
+        ('PS384', 'RSASSA-PSS signature using SHA-384 and MGF1 padding with '
+         'SHA-384'),
+        ('PS512', 'RSASSA-PSS signature using SHA-512 and MGF1 padding with '
+         'SHA-512'),
+    ], string='Algorithm', help='Algorithm used to sign the JSON Web Token.')
+    jwt_private_key = fields.Text(
+        string='Private Key',
+        help='Private key used for the JSON Web Token generation.')
+    jwt_public_key = fields.Text(
+        string='Public Key', compute='_compute_jwt_public_key',
+        help='Public key used for the JSON Web Token generation.')
+
+    @api.multi
+    def _load_private_key(self):
+        """ Load the client's private key into a cryptography's object instance
+        """
+        return load_pem_private_key(
+            str(self.jwt_private_key),
+            password=None,
+            backend=default_backend(),
+        )
+
+    @api.multi
+    @api.constrains('jwt_algorithm', 'jwt_private_key')
+    def _check_jwt_private_key(self):
+        """ Check if the private key's type matches the selected algorithm
+
+        This check is only performed for asymetric algorithms
+        """
+        for client in self:
+            algorithm_prefix = client.jwt_algorithm[:2]
+            if algorithm_prefix in ALGORITHMS and client.jwt_private_key:
+                private_key = client._load_private_key()
+
+                if not isinstance(
+                   private_key, ALGORITHMS[algorithm_prefix]):
+                    raise exceptions.ValidationError(
+                        _('The private key doesn\'t fit the selected '
+                          'algorithm!'))
+
+    @api.multi
+    def generate_private_key(self):
+        """ Generate a private key for ECDSA and RSA algorithm clients """
+        for client in self:
+            algorithm_prefix = client.jwt_algorithm[:2]
+            if algorithm_prefix not in ALGORITHMS:
```

Comment:
I replaced this by an `else: raise`, to warn the user if he clicks on the button with the wrong algorithm.